autorest/azure/token.go

```diff
@@ -301,14 +301,17 @@ func (spt *ServicePrincipalToken) refreshInternal(resource string) error {
 	}
 
 	var newToken Token
+	var aadError AADServiceError
 	err = autorest.Respond(resp,
 		autorest.WithErrorUnlessOK(),
-		autorest.ByUnmarshallingJSON(&newToken),
+		autorest.ByUnmarshallingSuccessandErrorJSON(&newToken, &aadError),
```

Comment:
This works. I assumed that it follows standard message handling, where responder respond calls previous responder respond. So autorest.ByUnmarshallingSuccessandErrorJSON would have called  autorest.WithErrorUnlessOK.